rpc/daemon.go

```diff
@@ -0,0 +1,56 @@
+package rpc
+
+import (
+	"net/http"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+
+	"github.com/athanorlabs/atomic-swap/cliutil"
+	"github.com/athanorlabs/atomic-swap/common"
+	"github.com/athanorlabs/atomic-swap/net"
+)
+
+// DaemonService handles general daemon operations
+type DaemonService struct {
+	server *Server
+	pb     ProtocolBackend
+}
+
+// NewDaemonService ...
+func NewDaemonService(server *Server, pb ProtocolBackend) *DaemonService {
+	return &DaemonService{
+		server,
+		pb,
+	}
+}
+
+// ShutdownRequest ...
+type ShutdownRequest struct{}
+
+// ShutdownResponse ...
+type ShutdownResponse struct{}
+
+// Shutdown swapd
+func (s *DaemonService) Shutdown(_ *http.Request, req *ShutdownRequest, resp *ShutdownResponse) error {
+	return s.server.Stop()
+}
+
+// VersionRequest ...
+type VersionRequest struct{}
+
+// VersionResponse ...
+type VersionResponse struct {
+	SwapdVersion    string             `json:"swapdVersion" validate:"required"`
+	P2PVersion      string             `json:"p2pVersion" validate:"required"`
+	Env             common.Environment `json:"env" validate:"required"`
+	SwapCreatorAddr ethcommon.Address  `json:"swapCreatorAddress" validate:"required"`
+}
+
+// Version returns version & misc info about swapd and its dependencies
+func (s *DaemonService) Version(_ *http.Request, req *VersionRequest, resp *VersionResponse) error {
```

Comment:
```suggestion
func (s *DaemonService) Version(_ *http.Request, _ *interface{}, resp *VersionResponse) error {
```

similarly here

Reply:
Here's a unit test that you can add to the `daemon` package to test the RPC version call end-to-end. You'll need add the chainID to the protocol version for it to work:
```
// Tests the version and shutdown RPC methods
func TestRunSwapDaemon_RPC_Version(t *testing.T) {
	conf := createTestConf(t, tests.GetMakerTestKey(t))
	protocolVersion := fmt.Sprintf("%s/%d", net.ProtocolID, conf.EthereumClient.ChainID())
	timeout := time.Minute
	ctx := launchDaemons(t, timeout, conf)

	c := rpcclient.NewClient(ctx, fmt.Sprintf("http://127.0.0.1:%d", conf.RPCPort))
	versionResp, err := c.Version()
	require.NoError(t, err)

	require.Equal(t, conf.EnvConf.Env, versionResp.Env)
	require.NotEmpty(t, versionResp.SwapdVersion)
	require.Equal(t, conf.EnvConf.SwapCreatorAddr, versionResp.SwapCreatorAddr)
	require.Equal(t, protocolVersion, versionResp.P2PVersion)
}
```
Interesting note.  I was playing around with @noot's suggestion.  You have to use `_ *any` on the request type.  If you try to use `_ any`, you'll get the error `rpc: can't find method "daemon.Version"`.